app/models/miq_ae_datastore.rb

```diff
@@ -170,17 +170,16 @@ def self.reset_to_defaults
   end
 
   def self.reset_domains_from_legacy_directory
-    domain_files = DATASTORE_DIRECTORY.join('*', MiqAeDomain::DOMAIN_YAML_FILENAME)
-    Dir.glob(domain_files).each do |domain_file|
-      domain_name = File.basename(File.dirname(domain_file))
-      reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant)
-    end
+    reset_domain(DATASTORE_DIRECTORY, domain_names, Tenant.root_tenant)
   end
 
   def self.reset_domains_from_vmdb_plugins
-    datastores_for_reset.each do |domain|
-      reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant)
-    end
+    datastores_for_reset.each { |domain| reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant) }
+  end
+
+  def self.domain_names
+    files = DATASTORE_DIRECTORY.join('*', MiqAeDomain::DOMAIN_YAML_FILENAME)
```

Comment:
I would suggest that we make two private methods `plugin_domain_names` and `legacy_domain_names` and this method would becomes `plugin_domain_names + legacy_domain_names`